components/pd_client/src/util.rs

```diff
@@ -43,6 +43,38 @@ const MAX_RETRY_DURATION: Duration = Duration::from_secs(10);
 const GLOBAL_RECONNECT_INTERVAL: Duration = Duration::from_millis(100); // 0.1s
 pub const REQUEST_RECONNECT_INTERVAL: Duration = Duration::from_secs(1); // 1s
 
+pub struct TargetInfo {
+    target_url: String,
+    via: String,
+}
+
+impl TargetInfo {
+    fn new(target_url: String, via: &str) -> TargetInfo {
+        TargetInfo {
+            target_url,
```

Comment:
Is this trimmed?

Reply:
It's not. And it's expected not to trim as PD only accepts url currently.